File: Problemas sobre RSQ y LIS/ilya.py
from sys import stdin


def fill_array(sequence: list[int]) -> list[int]:
    """
    Llena un array con la cantidad de coincidencias de la secuencia por posición;
    utilizando programación dinámica
    Params:
        sequence: list[int]
            secuencia de string leídas
    Returns:
        equals: list[int]
            array con la cantidad de coincidencias de la secuencia leída por posición
    """
    n = len(sequence)
    equals = [0] * n
    for i in range(n - 2, -1, -1):
        equals[i] = equals[i + 1]
        if sequence[i] == sequence[i + 1]:
            equals[i] += 1
    return equals


sequence = stdin.readline().strip()  # Se lee el string
queries = int(stdin.readline().strip())  # Se lee la cantidad de queries
result_array = fill_array(list(sequence))
for _ in range(queries):
    indexes = stdin.readline().strip().split(" ")
    print(result_array[int(indexes[0]) - 1] - result_array[int(indexes[1]) - 1])

File: Problemas sobre RSQ y LIS/ilya.js
const readline = require('readline');

function fillArray(sequence) {
    const n = sequence.length;
    const equals = new Array(n).fill(0);
    for (let i = 1; i < n; i++) {
        equals[i] = equals[i - 1];
        if (sequence[i] === sequence[i - 1]) {
            equals[i] += 1;
        }
    }
    return equals;
}

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

let input = [];
rl.on('line', (line) => {
    input.push(line.trim());
});

rl.on('close', () => {
    const sequence = input[0];
    const queries = parseInt(input[1], 10);
    const resultArray = fillArray(sequence);

    for (let i = 2; i < 2 + queries; i++) {
        const [li, ri] = input[i].split(" ").map(Number);
        console.log(resultArray[ri - 1] - resultArray[li - 1]);
    }
});